book/src/plumbing/db_lifetime.md

```diff
@@ -0,0 +1,238 @@
+# The `'db` lifetime
+
+[Tracked](./tracked_structs.md) and interned structs are both declared with a `'db` lifetime.
+This lifetime is linked to the `db: &DB` reference used to create them.
+The `'db` lifetime has several implications:
+
+* It ensures that the user does not create a new salsa revision while a tracked/interned struct is in active use. Creating a new salsa revision requires modifying an input which requires an `&mut DB` reference, therefore it cannot occur during `'db`.
+    * The struct may not even exist in the new salsa revision so allowing access would be confusing.
+* It permits the structs to be implemented using a pointer rather than a `salsa::Id`, which in turn means more efficient field access (no read locks required).
+
+This section discusses the unsafe code used for pointer-based access along with the reasoning behind it. To be concrete, we'll focus on tracked structs -- interned structs are very similar.
+
+## A note on UB
+
+When we say in this page "users cannot do X", we mean without Undefined Behavior (e.g., by transmuting integers around etc).
+
+## Proof obligations
+
+Here is a typical sequence of operations for a tracked struct along with the user operations that will require us to prove unsafe assertions:
+
+* A tracked function `f` executes in revision R0 and creates a tracked struct with `#[id]` fields `K` for the first time.
+    * `K` will be stored in the interning hashmap and mapped to a fresh identifier `id`.
+    * The identifier `id` will be used as the key in the `StructMap` and point to a freshly created allocation `alloc : Alloc`.
+    * A `ts: TS<'db>` is created from the raw pointer `alloc` and returned to the user.
```

Comment:
It's not entirely clear to readers what `ts` refers to. There's an explanation below in the definitions section. Maybe add it to the sentence in the first bullet point like this

> creates a tracked struct (ts) with `#[id]` fields